src/index.ts

```diff
@@ -613,17 +625,24 @@ async function ensureBackwardsCompatibility(components: Components, config: Conf
   // - Root read key of the filesystem: IndexedDB → localforage → readKey
   // - Authenticated username: IndexedDB → localforage → webnative.auth_username
 
-  const [ migK, migV ] = [ "migrated", "true" ]
+  const [ migK, migV ] = [ "migrated", VERSION ]
+  const currentVersion = Semver.fromString(VERSION)
+  if (!currentVersion) throw new Error("The webnative VERSION should be a semver string")
 
   // If already migrated, stop here.
-  const migrationOccurred = await components.storage.getItem(migK) === migV
+  const migrationOccurred = await components.storage
+    .getItem(migK)
+    .then(v => typeof v === "string" ? Semver.fromString(v) : null)
+    .then(v => v && Semver.isBiggerThanOrEqualTo(v, currentVersion))
+
   if (migrationOccurred) return
 
   // Only try to migrate if environment supports indexedDB
   if (!self.indexedDB) return
 
   // Migration
-  const keystoreDB = await bwOpenDatabase("keystore")
+  const existingDatabases = (await self.indexedDB.databases()).map(db => db.name)
```

Comment:
Seeing this error in Firefox:

<img width="958" alt="idx-db" src="https://user-images.githubusercontent.com/7957636/204662950-2e6ef73b-ee23-49b7-9dd7-474398de3832.png">

It looks like `indexedDB.databases` is not supported in Firefox: https://developer.mozilla.org/en-US/docs/Web/API/IDBFactory/databases

Reply:
Thanks! Added a fallback for Firefox and older browsers.